pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java

```diff
@@ -79,6 +79,21 @@ public Dictionary getDictionary() {
     throw new UnsupportedOperationException();
   }
 
+  protected String notBytesErrorMessage(TransformFunction transformFunction) {
+    return String.format("The argument must be of type %s , but was %s",
+        DataType.BYTES,
+        transformFunction.getResultMetadata().getDataType()
+    );
```

Comment:
nit
```suggestion
    return notBytesErrorMessage(transformFunction, "first");
```

Reply:
@walterddr sounds good to me. I only made that extra function as the current error message doesn't mention the position of the argument if there's only one.